feature-libs/product-configurator/rulebased/components/attribute/types/base/configurator-attribute-single-selection-base.component.ts

```diff
@@ -0,0 +1,158 @@
+import { ConfiguratorAttributeBaseComponent } from './configurator-attribute-base.component';
+import { Directive, EventEmitter, Input, Output } from '@angular/core';
+import { map } from 'rxjs/operators';
+import { BehaviorSubject } from 'rxjs';
+import { FormControl } from '@angular/forms';
+import { Configurator } from '../../../../core/model/configurator.model';
+import { ConfigFormUpdateEvent } from '../../../form/configurator-form.event';
+import { ConfiguratorAttributeQuantityComponentOptions } from '../../quantity/configurator-attribute-quantity.component';
+import { ConfiguratorAttributeQuantityService } from '../../quantity/configurator-attribute-quantity.service';
+import { ConfiguratorPriceComponentOptions } from '../../../price/configurator-price.component';
+
+@Directive()
+// eslint-disable-next-line @angular-eslint/directive-class-suffix
+export abstract class ConfiguratorAttributeSingleSelectionBaseComponent extends ConfiguratorAttributeBaseComponent {
+  loading$ = new BehaviorSubject<boolean>(false);
+
+  @Input() attribute: Configurator.Attribute;
+  @Input() ownerKey: string;
+  @Output() selectionChange = new EventEmitter<ConfigFormUpdateEvent>();
+
+  // TODO(#11681): make quantityService a required dependency
+  /**
+   * default constructor
+   * @param {ConfiguratorAttributeQuantityService} quantityService
+   */
+  // eslint-disable-next-line @typescript-eslint/unified-signatures
+  constructor(quantityService: ConfiguratorAttributeQuantityService);
+
+  /**
+   * @deprecated since 3.3
+   */
+  constructor();
+
+  constructor(
+    protected quantityService?: ConfiguratorAttributeQuantityService
+  ) {
+    super();
+  }
```

Comment:
Have to handle breaking changes in children (drop-down, and radio-button), because some children are new classes, there are no breaking changes issues for them. So, in here, it should be:
```
  // TODO(#11681): make quantityService a required dependency
  constructor(
    protected quantityService?: ConfiguratorAttributeQuantityService
  ) {
    super();
  }
```

Reply:
I am not very sure about this, any thought?

Reply:
Picked!

Reply:
Done!